### PR TITLE
Add autoconnect support for hidden WiFi networks

### LINKED ISSUE #31651: [$100 bounty] Hidden WiFi network entries not persistent.
### Describe the bug

When manually entering a hidden WiFi network, the connection succeeds, but will not automatically reconnect.

1. Enter SSID and password for a hidden WiFi network.
2. Go on a drive until that network connection is lost.
3. Return to where the connection was first established.
4. Observe that WiFi does not automatically reconnect to the hidden network.
5. Observe that repeating the first step still works.

### Provide a route where the issue occurs

N/A

### openpilot version

0.9.6

### Additional info

2021 Nissan Leaf SV (40 KWh) with Comma 3X.

I only got my C3X a month ago (I'm a noob), and my free Connect trial expired last week (no route to share).

### PR DESCRIPTION
**Pull request by BBBmau**
Fixes #31651

### CODE REVIEW

**Comment by royjr** [comment]
Didn't work for me until I used wifi/hidden=true. Confirmed on both open and secured hidden networks.

**Comment by adeebshihadeh** [comment]
How did you test this? Isn't it the default?

**Comment by BBBmau** [comment]
You're right, after looking at docs the default is `true` @adeebshihadeh 

@royjr looked into it also and found that by default is set to `false` and is what prevents the connection from autoconnecting for hidden networks.

https://networkmanager.dev/docs/api/latest/settings-802-11-wireless.html

hidden

	

boolean

	

FALSE
```
If TRUE, indicates that the network is a non-broadcasting network that hides its SSID. This works both in infrastructure and AP mode. In infrastructure mode, various workarounds are used for a more reliable discovery of hidden networks, such as probe-scanning the SSID. However, these workarounds expose inherent insecurities with hidden SSID networks, and thus hidden SSID networks should be used with caution. In AP mode, the created network does not broadcast its SSID. Note that marking the network as hidden may be a privacy issue for you (in infrastructure mode) or client stations (in AP mode), as the explicit probe-scans are distinctly recognizable on the air.
```

**Comment by MarcoTheDingo** [comment]
Are we setting hidden=true for all networks it attempts to connect to, or only ones marked as Hidden from networking.cc? There could be potential security risks if we use hidden=true for all networks.

**Comment by BBBmau** [comment]
@MarcoTheDingo I'll add a condition for this 👍🏼

**Comment by BBBmau** [comment]
latest commit prevents `connection[wireless][hidden]` from being set to true if not a hidden network.

How did you go about testing it locally yourself? I can't go out and test it myself on my device right now so it would be nice to have a setup in times like this. @royjr

**Comment by royjr** [comment]
nothing special, went out and tested in car

**Comment by MarcoTheDingo** [comment]
I tested it with a dummy hidden network I set up on my UDM-SE / U6 Lite. Seemed to join right away after a reboot. Nice job!

File: selfdrive/ui/qt/network/wifi_manager.h [hunk 1]
```diff
@@ -52,7 +52,7 @@ class WifiManager : public QObject {
   std::optional<QDBusPendingCall> activateWifiConnection(const QString &ssid);
   NetworkType currentNetworkType();
   void updateGsmSettings(bool roaming, QString apn, bool metered);
-  void connect(const Network &ssid, const QString &password = {}, const QString &username = {});
+  void connect(const Network &ssid, const bool is_hidden, const QString &password = {}, const QString &username = {});
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/network/wifi_manager.h:55]: make this default to false

Comment (BBBmau) [reply selfdrive/ui/qt/network/wifi_manager.h]: done

**Comment by FlyMyPG** [comment]
Questions (that may reveal my ignorance of NetworkManager):

1) Will this change set the hidden flag to true for any manually entered SSID, no matter if the network is actually hidden or not? For example, if I enter an SSID/password for a visible network when I'm out of range.

2) Is the hidden flag ever automatically set to false if a previously hidden network becomes visible? In my case, my wlan is hidden by default, but I occasionally need to briefly make it visible to add some brands of rather stupid home automation devices.

3) Should the user be able to manually set the hidden flag in the OP UI whenever an SSID is entered manually?

**Comment by BBBmau** [comment]
@FlyMyPG this is explicitly for hidden networks since you have to enter the UI for connecting to them. https://github.com/commaai/openpilot/pull/30808

by default it's set to false when you connect to a normal network, so in your case you will be good!

**Pull request merged**